compiler/src/iree/compiler/Codegen/Common/ConvertBf16ToUInt16Buffers.cpp

```diff
@@ -225,6 +226,29 @@ struct ConvertMemRefStore final : OpConversionPattern<memref::StoreOp> {
   }
 };
 
+struct ConvertAmdgpuFatRawBufferCast final
+    : OpConversionPattern<amdgpu::FatRawBufferCastOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(amdgpu::FatRawBufferCastOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Type newTy = getTypeConverter()->convertType(op.getType());
+    if (!newTy)
+      return rewriter.notifyMatchFailure(
+          op->getLoc(),
+          llvm::formatv("failed to convert memref type: {}", op.getType()));
+
+    auto newOp = rewriter.replaceOpWithNewOp<amdgpu::FatRawBufferCastOp>(
+        op, newTy, adaptor.getSource(), adaptor.getValidBytes(),
+        adaptor.getCacheSwizzleStride(), adaptor.getBoundsCheck(),
+        adaptor.getResetOffset());
+    LLVM_DEBUG(llvm::dbgs() << "Bf16Emulation: new op: " << newOp << "\n");
```

Comment:
I dont know if we want to add such debug prints... The dialect conversion already does a pretty good job of debug output.